core/src/test/java/org/springframework/security/authorization/AuthorityAuthorizationManagerTests.java

```diff
@@ -91,6 +93,18 @@ public void hasAnyAuthorityWhenContainNullThenException() {
 				.withMessage("authorities cannot contain null values");
 	}
 
+	@Test
+	public void hasAnyAuthorityWhenUserHasCustomAuthorityThenGrantedDecision() {
```

Comment:
Can you also provide a test for when the `Authentication` has the wrong authorities but uses a custom implementation?

Reply:
Done.